src/pages/LogInWithShortLivedTokenPage.js

```diff
@@ -69,16 +65,20 @@ class LogInWithShortLivedTokenPage extends Component {
         }
 
         signInWithShortLivedToken(accountID, email, shortLivedToken, encryptedAuthToken);
-        this.setState({hasRun: true});
     }
 
     componentDidUpdate() {
-        if (this.state.hasRun || !this.props.betas) {
+        const email = lodashGet(this.props.route.params, 'email', '');
+        if (!this.props.betas || !this.props.session.authToken || email !== this.props.session.email) {
```

Comment:
Do we still need to wait for `this.props.betas`?

Reply:
Okay, np. I'll clean this up and retest.

Reply:
If we tried to navigate here and the betas weren't loaded then the modal would just close 😅